applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py

```diff
@@ -0,0 +1,44 @@
+import KratosMultiphysics
+import KratosMultiphysics.FluidDynamicsApplication
+import KratosMultiphysics.RomApplication as romapp
+from KratosMultiphysics.RomApplication import python_solvers_wrapper_rom as solver_wrapper
+from KratosMultiphysics.FluidDynamicsApplication.fluid_dynamics_analysis import FluidDynamicsAnalysis
+
+import json
+
+class FluidDynamicsAnalysisROM(FluidDynamicsAnalysis):
+
+    def __init__(self,model,project_parameters):
+        super().__init__(model,project_parameters)
+
+    #### Internal functions ####
+    def _CreateSolver(self):
+        """ Create the Solver (and create and import the ModelPart if it is not alread in the model) """
+        ## Solver construction
+        with open('RomParameters.json') as rom_parameters:
+            rom_settings = KratosMultiphysics.Parameters(rom_parameters.read())
+            self.project_parameters["solver_settings"].AddValue("rom_settings", rom_settings["rom_settings"])
+        return solver_wrapper.CreateSolverByParameters(self.model, self.project_parameters["solver_settings"],self.project_parameters["problem_data"]["parallel_type"].GetString())
+
+    def _GetSimulationName(self):
+        return "::[ROM Simulation]:: "
+
+    def ModifyInitialGeometry(self):
```

Comment:
It is a bit counterintuitive to me to do this in the `ModifyInitialGeometry` function. Could it be done in `ModifyAfterSolverInitialize`? (not sure if you need your solver to have this values set before calling the `Initialize`. If this is the case, I think we could propose introducing a ` ModifyBeforeSolverInitialize` to the `AnalysisStage`.

Reply:
I have no objection. I require the nodes to have their ROM_BASIS and the Elements to have their ROM_WEIGHT before the SetSetUpDofSet method of the RomBuilder&Solver is called. I changed it as suggested

Reply:
I think this is a better usage of the `AnalysisStage` methods. Thanks!